### PR TITLE
ISPN-11532 HotRod Worker Thread Pool

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-11532
https://issues.redhat.com/browse/ISPN-11531
https://issues.redhat.com/browse/ISPN-11755

### CODE REVIEW

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java [hunk 1]
```diff
@@ -104,23 +101,16 @@ public void start(C configuration, EmbeddedCacheManager cacheManager) {
...
-                     throw new IllegalLifecycleStateException("Server has been stopped");
-                  else
-                     super.rejectedExecution(r, e);
-               }
-            });
-
-      manageableThreadPoolExecutorService = new ManageableThreadPoolExecutorService(executor);
+      executor = bcr.getComponent(KnownComponentNames.BLOCKING_EXECUTOR, ExecutorService.class).running();
+
+      if (executor instanceof BlockingTaskAwareExecutorServiceImpl) {
```
Comment (pruivo) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java:106]: when is this `true`? 
I think we only create `LazyInitializingBlockingTaskAwareExecutorService`. It is already a `ManageableExecutorService` so it can be used directly without more wrapping.

Comment (wburns) [reply server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java:106]: Hrmm, I had an issue in testing, maybe it isn't a problem anymore. Let me try again.

Comment (wburns) [reply server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java]: I have removed the if, if CI passes then it should be good ;)

**Comment by pruivo** [comment]
triggered another CI run...

**Comment by wburns** [comment]
It looks like there is a shutdown issue I need to resolve.

**Comment by wburns** [comment]
Okay it should be fixed now.

**Comment by pruivo** [comment]
integrated! thanks @wburns !
